src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringWithoutLastElement = '';
+    const lastElement = this[this.length - 1];
+    let x = separator;
+
+    if (x === null) {
+      x = 'null';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
```

Comment:
just check if this[i] exists and check if it's not the last element
and if it's not add separator, otherwise return result string